File: b1-programming-basics-python/03b-conditional-statements-advanced-exercise/09_ski _trip.py
days = int(input())
room_type = input()
grade = input()
price = 0
room_for_one_person = 18
apartment = 25
president_apartment = 35
nights = days - 1
if room_type == "apartment":
    if days < 10:
        price = apartment * nights - (apartment * nights * 0.3)
    elif 10 < days < 15:
        price = apartment * nights - (apartment * nights * 0.35)
    else:
        price = apartment * nights - (apartment * nights * 0.5)
elif room_type == "room for one person":
    price = room_for_one_person * nights
elif days < 10:
    price = president_apartment * nights - (president_apartment * nights * 0.1)
elif 10 < days < 15:
    price = president_apartment * nights - (president_apartment * nights * 0.15)
else:
    price = president_apartment * nights - (president_apartment * nights * 0.2)
if grade == "positive":
    price += price * 0.25
else:
    price -= price * 0.10
print(f"{price:.2f}")

File: a1-softuni-programming-basics-javascript/03b-conditional-statements-advanced-exercise/09-ski-trip.js
function skiTrip(input) {
  let nights = Number(input[0]) - 1;
  let roomType = input[1].toLowerCase();
  let happyCondition = input[2].toLowerCase();
  let sum = 0;
  let discount = 0;
  let pricePerNight = 0;

  switch (roomType) {
    case `room for one person`:
      pricePerNight = 18;
      break;
    case `apartment`:
      pricePerNight = 25;
      if (nights < 10) {
        discount += 0.3;
      } else if (nights >= 10 && nights <= 15) {
        discount += 0.35;
      } else {
        discount += 0.5;
      }
      break;
    case `president apartment`:
      pricePerNight = 35;
      if (nights < 10) {
        discount += 0.1;
      } else if (nights >= 10 && nights <= 15) {
        discount += 0.15;
      } else {
        discount += 0.2;
      }

      break;

    default:
      break;
  }
  sum = nights * pricePerNight;
  if (discount > 0) {
    sum -= sum * discount;
  }

  if (happyCondition === "positive") {
    sum += sum * 0.25;
  } else {
    sum -= sum * 0.1;
  }

  console.log(sum.toFixed(2));
}
